### PR TITLE
backporting the what-service-mesh page for the intro

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
This PR backports the changes introduced in PR #11962  to the 1.10.x release branch

### CODE REVIEW

**Review by trujillo-adam — APPROVED** [review]

File: website/content/docs/k8s/helm.mdx [hunk 1]
```diff
@@ -689,20 +679,6 @@ Use these links to navigate to a particular top-level stanza.
...
-  - `containerSecurityContext` ((#v-client-containersecuritycontext)) (`map`) - The container securityContext for each container in the client pods. In
-    addition to the Pod's SecurityContext this can
-    set the capabilities of processes running in the container and ensure the
-    root file systems in the container is read-only.
-
-    - `client` ((#v-client-containersecuritycontext-client)) (`map`) - The consul client agent container
-
-    - `aclInit` ((#v-client-containersecuritycontext-aclinit)) (`map`) - The acl-init initContainer
-
-    - `tlsInit` ((#v-client-containersecuritycontext-tlsinit)) (`map`) - The tls-init initContainer
```
Comment (lkysow) [inline website/content/docs/k8s/helm.mdx:703]: These lines too

**Review by lkysow — APPROVED** [review]

File: website/content/docs/k8s/helm.mdx (hunk 1 above)
Comment (karl-cardenas-coding) [reply website/content/docs/k8s/helm.mdx:703]: Thanks for the sanity check @lkysow . Adding the now.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/576336.

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 8b609a9c2dd7262af5958c4d3b2817b643beb057 onto `release/1.10.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/576336)

**Pull request merged**